contracts/deposit-contract/sendDepositTx/sendDeposits_test.go

```diff
@@ -134,28 +88,27 @@ func TestEndtoEndDeposits(t *testing.T) {
 
 	query := ethereum.FilterQuery{
 		Addresses: []common.Address{
-			web3Service.depositContractAddress,
+			testAcc.ContractAddr,
 		},
 	}
 
-	logs, err := testAcc.Backend.FilterLogs(web3Service.ctx, query)
+	logs, err := testAcc.Backend.FilterLogs(context.Background(), query) //context backgrround ?
 	if err != nil {
 		t.Fatalf("Unable to retrieve logs %v", err)
 	}
 	if len(logs) == 0 {
 		t.Fatal("no logs")
 	}
 
-	web3Service.chainStarted = true
 
-	for _, log := range logs {
-		if err := web3Service.ProcessDepositLog(context.Background(), log); err != nil {
-			t.Fatal("Unable to process deposit log %v", err)
-		}
-	}
-	totalNumberOfDeposits := validatorsWanted * numberOfDeposits
-	pendingDeposits := web3Service.depositCache.PendingDeposits(context.Background(), nil /*blockNum*/)
-	if len(pendingDeposits) != int(totalNumberOfDeposits) {
-		t.Errorf("Unexpected number of deposits. Wanted %v deposits, got %+v", int(totalNumberOfDeposits), pendingDeposits)
-	}
+    // TODO to check deposit count
```

Comment:
@nisdas I assume, i have to check deposit count that deposit contract has been processed, what else should I include in my test?

Reply:
yes check the deposit count, validate deposit data and ensure it is accurate and the same data you sent over in a transaction, validate merkle root of trie in the deposit contract